### PR TITLE
Client certificate support

### PR DESCRIPTION
**Pull request by mike-schiller**
Allows for optional client certificate / key to be used when connecting to a Crate server via https. This is useful if Crate is running behind a revers proxy that validates client certificates or for modified versions of Crate that support client certificate validation.

### CODE REVIEW

File: src/crate/client/http.py [hunk 1]
```diff
@@ -127,7 +128,7 @@ def __init__(self, servers=None, timeout=None, ca_cert=None,
         self._active_servers = servers
         self._inactive_servers = []
         self._http_timeout = timeout
-        pool_kw = {}
+        pool_kw = {'cert_file':cert_file, 'key_file':key_file}
```
Comment (mfussenegger) [inline src/crate/client/http.py]: does it really make sense to add them to the dict even if they're None?

File: src/crate/client/connection.py [hunk 2]
```diff
@@ -28,7 +28,9 @@
 
 class Connection(object):
     def __init__(self, servers=None, timeout=None, client=None,
-                 verify_ssl_cert=False, ca_cert=None, error_trace=False):
+                 verify_ssl_cert=False, ca_cert=None,
+                 cert_file=None, key_file=None,
```
Comment (mfussenegger) [inline src/crate/client/connection.py]: could you add the args as last params in order to not break backward compatibility for people who might have used positional arguments?

**Comment by mfussenegger** [comment]
Sorry for the late response. Seems like this PR was overlooked.

**Comment by chaudum** [comment]
hi @mike-schiller 

do you want to add the changes @mfussenegger suggested?
also, could you add an entry to the CHANGES.txt file (in the "Unreleased" section)

thanks,
christian

ps: did you sign a [Contributor Agreement](https://crate.io/community/contribute/agreements/) already?

**Comment by chaudum** [comment]
@mike-schiller updates on this one?

**Comment by mfussenegger** [comment]
We've released 0.15 which contains client certificate support, so I'm closing this.

**Pull request closed without merging**